yellowbrick/features/rankd.py

```diff
@@ -240,27 +275,173 @@ def rank(self, X, algorithm=None):
                 "'{}' is unrecognized ranking method".format(algorithm)
             )
 
+        # Extract matrix from dataframe if necessary
+        if is_dataframe(X):
```

Comment:
Yeh, we're dealing with that challenge in a lot of places. Let's not worry about it for this PR but make an issue regarding it so that we can explore it in the future. I'm more keen to get this merged sooner rather than delay on figuring stuff like this out completely!